---
title: C6255
ms.date: 11/04/2016
ms.prod: visual-studio-dev15
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C6255
helpviewer_keywords:
- C6255
ms.assetid: bb6430b2-782a-4410-a8e1-609df06007de
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: 0692ee1523f595054baa623a7fdd4df13ac2bc3a
ms.sourcegitcommit: 240c8b34e80952d00e90c52dcb1a077b9aff47f6
ms.translationtype: MT
ms.contentlocale: pt-BR
ms.lasthandoff: 10/23/2018
ms.locfileid: "49819063"
---
# <a name="c6255"></a>C6255
Aviso C6255: alloca indicou falha, gerando uma exceção de estouro de pilha. Considere utilizar malloca em vez disso

 Este aviso indica que uma chamada para `_alloca` foi detectada fora do tratamento de exceção de local. `_alloca` deve sempre ser chamado de dentro do intervalo protegido de um manipulador de exceção porque pode lançar uma exceção de estouro de pilha em caso de falha. Se possível, em vez de usar `_alloca`, considere o uso `_malloca` que é uma versão mais segura de `_alloca`.

## <a name="example"></a>Exemplo
 O código a seguir gera este aviso, pois `_alloca` possa gerar exceção:

```
#include <windows.h>

void f( )
{
  void *p = _alloca(10);
  // code ...
}
```

 Para corrigir este aviso, use `_malloca` e adicione o manipulador de exceção, conforme mostrado no código a seguir:

```
#include <windows.h>
#include <malloc.h>

void f( )
{
  void *p;
  int errcode;
  __try
  {
    p = _malloca(10);
    // code...
    _freea(p);
  }
  __except( (GetExceptionCode() == STATUS_STACK_OVERFLOW ) ?
              EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )
  {
    errcode = _resetstkoflw();
    // code ...
  }
}
```

## <a name="see-also"></a>Consulte também
 [_malloca](/cpp/c-runtime-library/reference/malloca)